text/0317-third-party-dependency-management.md

```diff
@@ -0,0 +1,129 @@
+---
+rfc pr: [#319](https://github.com/aws/aws-cdk-rfcs/pull/319)
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/317
+---
+
+# CDK third-party dependencies management
+
+The Cloud Development Kit depends on a large collection of open-source third-party libraries.
+While leveraging existing libraries that implement functionality needed by the project is obviously immensely useful,
+it also requires correctly managing those dependencies, in particular:
+
+1. Quickly reacting when a security vulnerability is discovered in a version of a dependency used by the project.
+
+2. Protecting ourselves against [supply chain attacks](https://en.wikipedia.org/wiki/Supply_chain_attack),
+  like [typosquatting](https://www.darkreading.com/vulnerabilities---threats/beware-the-package-typosquatting-supply-chain-attack/a/d-id/1340383),
+  malicious package takeovers, etc.
+
+## Current situation
+
+### Dependencies
+
+Right now, our released packages bundle all of their dependencies.
+
+#### CLIs
+
+In our CLI packages (`aws-cdk`, `cdk-assets`, etc.) that are NodeJS-only,
+we have a very large dependency closure
+(around 180 packages - [see here](https://npmgraph.js.org/?q=aws-cdk)
+for the details).
+
+At build time of these packages,
+we generate an `npm-shrinkwrap.json` file with the exact version of each dependency
+(including transitive ones).
+This means a customer installing those packages in a given version will always get all of their dependencies in a specific version
+(the one defined in `npm-shrinkwrap.json`) as well.
+
+#### Framework
+
+In our framework packages, we have a much smaller dependency closure,
+roughly equal to:
```

Comment:
Why the "roughly" here? This set is small enough to fully and accurately enumerate, no?

Is it because of the specific versions?

Reply:
Roughly, because one second after I post this, someone can add a new dependency to our closure.

But fine, removed the "roughly".